src/goto-programs/adjust_float_expressions.cpp

```diff
@@ -72,14 +75,15 @@ static bool have_to_adjust_float_expressions(const exprt &expr)
   return false;
 }
 
-/// This adds the rounding mode to floating-point operations, including those in
-/// vectors and complex numbers.
 void adjust_float_expressions(exprt &expr, const exprt &rounding_mode)
 {
   if(!have_to_adjust_float_expressions(expr))
     return;
 
   // recursive call
+  // note that we do the recursion twice here; once in `have_to_adjust_float_expressions`
+  // and once here. Presumably this is to avoid breaking sharing (calling the non-const
+  // operands() function breaks sharing)
```

Comment:
Yes, it is done to avoid breaking sharing. With the depth iterator a different approach may be possible (cf. the changes in 430d402046177208ee015c95e4e87e49ba2a7b20).